Project-Aurora/Project-Aurora/Settings/Layers/RazerLayerHandler.cs

```diff
@@ -0,0 +1,316 @@
+            _mousepadColors = new Color[16];
+
+            _instances++;
+            Global.logger.Debug("RazerLayerHandler intance count: {0}", _instances);
+            if (_instances > 0 && _manager == null)
+            {
+                if (!RzHelper.IsSdkVersionSupported(RzHelper.GetSdkVersion()))
+                {
+                    Global.logger.Warn("Currently installed razer sdk version \"{0}\" is not supported!", RzHelper.GetSdkVersion());
+                    return;
+                }
+
+                try
+                {
+                    _manager = new RzManager()
+                    {
+                        KeyboardEnabled = true,
+                        MouseEnabled = true,
+                        MousepadEnabled = true,
+                        AppListEnabled = true,
+                    };
+
+                    Global.logger.Info("RzManager loaded successfully!");
+                }
+                catch (Exception e)
+                {
+                    Global.logger.Fatal(e, "RzManager failed to load!");
+                }
+            }
+
+            if (_manager != null)
+            {
+                _manager.DataUpdated += OnDataUpdated;
+
+                var appList = _manager.GetDataProvider<RzAppListDataProvider>();
+                appList.Update();
+                _currentAppExecutable = appList.CurrentAppExecutable;
+                _currentAppPid = appList.CurrentAppPid;
+
+                Loaded = true;
+            }
+        }
+
+        protected override UserControl CreateControl()
+        {
+            return new Control_RazerLayer(this);
+        }
+
+        private void OnDataUpdated(object s, EventArgs e)
+        {
+            if (!(s is AbstractDataProvider provider))
+                return;
+
+            provider.Update();
+
+            if (_isDumping)
+                DumpData(provider);
+
+            if (provider is RzKeyboardDataProvider keyboard)
+            {
+                for (var i = 0; i < keyboard.ZoneGrid.Height * keyboard.ZoneGrid.Width; i++)
+                    _keyboardColors[i] = keyboard.GetZoneColor(i);
+            }
+            else if (provider is RzMouseDataProvider mouse)
+            {
+                _mouseColor = mouse.GetZoneColor(55);
+            }
+            else if (provider is RzMousepadDataProvider mousePad)
+            {
+                for (var i = 0; i < mousePad.ZoneGrid.Height * mousePad.ZoneGrid.Width; i++)
+                    _mousepadColors[i] = mousePad.GetZoneColor(i);
+            }
+            else if (provider is RzAppListDataProvider appList)
+            {
+                _currentAppExecutable = appList.CurrentAppExecutable;
+                _currentAppPid = appList.CurrentAppPid;
+            }
+        }
+
+        public bool StartDumpingData()
+        {
+            var root = $@"{Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)}\Aurora\Logs";
```

Comment:
You should share a constant for this path as you've reused the hardcoded line elsewhere, makes it a bit annoying if you want to easily switch which directory it's using. 
Side note: We really should have a global constant for the settings directory as if we ever want to change that, we'd have to change a lot of code

Reply:
Added AppDataDirectory and LogsDirectory constants and replaced in all places it was used.